src/TransactionFetcher.ts

```diff
@@ -0,0 +1,35 @@
+import Facilitator from './Facilitator';
+import GraphClient from './GraphClient';
+
+/**
+ * The class fetches the transactions based on uts.
+ */
+export default class TransactionFetcher {
+  private readonly graphClient: GraphClient;
+
+  /**
+   * Constructor
+   * @param graphClient Graph client object.
+   */
+  constructor(graphClient: GraphClient) {
+    this.graphClient = graphClient;
+  }
+
+  /**
+   * Queries graph node.
+   *
+   * @param data Data received from subscription.
+   * @return Graph query response from graph node.
+   */
+  public async fetch(data: Record<string, object>): Promise<{data: object}> {
+    const entity = Object.keys(data)[0];
+    const query = Facilitator.getSubscriptionDetails().fetchQueries[entity];
+    // Fetch entity based on uts from uts model and update the variables object
+    // Current dummy value is 0
+    const variables = {
```

Comment:
I guess we would be adding pagination logic here and would return response only after fetching all batches.

Reply:
Yes implementation in progress for pagination 📟